From: Shawn Rutledge <shawn.rutledge@qt.io>
Subject: [PATCH] Fix system md4c 0.6 compatibility

Backport the importer changes from Qt commit
032643f99e090e0304ef935c2ee11c109d55c8a1 (Update md4c to 0.6.0).
https://github.com/qt/qtbase/commit/032643f99e090e0304ef935c2ee11c109d55c8a1

Keep Qt's supported GitHub dialect flags and prevent its front matter flag
from enabling md4c's unsupported footnotes extension.

diff --git a/src/gui/text/qtextmarkdownimporter.cpp b/src/gui/text/qtextmarkdownimporter.cpp
index 9fdb7af23473..cf6628de7909 100644
--- a/src/gui/text/qtextmarkdownimporter.cpp
+++ b/src/gui/text/qtextmarkdownimporter.cpp
@@ -51,8 +51,10 @@ static_assert(int(QTextMarkdownImporter::FeaturePermissiveAutoLinks) == MD_FLAG_
 static_assert(int(QTextMarkdownImporter::FeatureTasklists) == MD_FLAG_TASKLISTS);
 static_assert(int(QTextMarkdownImporter::FeatureNoHTML) == MD_FLAG_NOHTML);
 static_assert(int(QTextMarkdownImporter::DialectCommonMark) == MD_DIALECT_COMMONMARK);
+// We don't yet support all the features in MD_DIALECT_GITHUB
 static_assert(int(QTextMarkdownImporter::DialectGitHub) ==
-              (MD_DIALECT_GITHUB | MD_FLAG_UNDERLINE | QTextMarkdownImporter::FeatureFrontMatter));
+              (MD_FLAG_PERMISSIVEAUTOLINKS | MD_FLAG_TABLES | MD_FLAG_STRIKETHROUGH | MD_FLAG_TASKLISTS
+                | MD_FLAG_UNDERLINE | QTextMarkdownImporter::FeatureFrontMatter));
 
 // --------------------------------------------------------
 // MD4C callback function wrappers
@@ -167,7 +169,7 @@ void QTextMarkdownImporter::import(const QString &markdown)
 {
     MD_PARSER callbacks = {
         0, // abi_version
-        unsigned(m_features),
+        unsigned(m_features) & unsigned(0xFFEFFFFF), // exclude MD_FLAG_FOOTNOTES: unsupported in Qt
         &CbEnterBlock,
         &CbLeaveBlock,
         &CbEnterSpan,
